### PR TITLE
Rabbit docs updates

### PR DESCRIPTION
**Pull request by bording**
This cleans up the snippets/samples project files, adds documentation around the new `UseDurableExchangesAndQueues` setting, and fixes up a few other miscellaneous things.

@Particular/rabbitmq-transport-maintainers Please review!

### CODE REVIEW

File: Snippets/Rabbit/Rabbit_All/Rabbit_All.csproj [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
     <TargetFrameworks>net462</TargetFrameworks>
     <LangVersion>7.1</LangVersion>
   </PropertyGroup>
+
   <ItemGroup>
+    <ProjectReference Include="..\..\Common\Common.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.*" />
```
Comment (adamralph) [inline Snippets/Rabbit/Rabbit_All/Rabbit_All.csproj]: FMI, how was this working before without this?

File: nservicebus/messaging/non-durable-messaging.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ Note: When using non transactional queues an endpoint has to be [configured to n
 
 ### RabbitMQ
 
-Non-durable messages are sent using the RabbitMQ non-persistent delivery mode, which means the messages are not persisted to disk by the broker. If durable messaging has been disabled globally, the exchanges and queues created by the broker will be declared as non-durable as well. If the broker is restarted, all non-durable exchanges and queues will be removed along with any messages in those queues.
+partial: rabbitmq
```
Comment (adamralph) [inline nservicebus/messaging/non-durable-messaging.md]: The partial is referred to as `rabbitmq`, but the file names contain `rabbitmq_core`. Will this work as expected?

File: Snippets/Rabbit/Rabbit_All/Rabbit_All.csproj (hunk 1 above)
Comment (bording) [reply Snippets/Rabbit/Rabbit_All/Rabbit_All.csproj]: Without that, the tests weren't showing up in VS at at all.

File: nservicebus/messaging/non-durable-messaging.md (hunk 2 above)
Comment (bording) [reply nservicebus/messaging/non-durable-messaging.md]: Notice the underscore separating them. That means the partial is named `rabbitmq`, and the component is `core`, which matches the component declaration on the page.

**Pull request merged**